src/blueprints/tv2/action-factories/tv2-remote-action-factory.ts

```diff
@@ -0,0 +1,120 @@
+import { PartInterface } from '../../../model/entities/part'
+import { PartActionType } from '../../../model/enums/action-type'
+import { Tv2BlueprintConfiguration } from '../value-objects/tv2-blueprint-configuration'
+import { Tv2SourceMappingWithSound } from '../value-objects/tv2-studio-blueprint-configuration'
+import { TimelineObject } from '../../../model/entities/timeline-object'
+import { Tv2PieceMetadata } from '../value-objects/tv2-metadata'
+import { PieceType } from '../../../model/enums/piece-type'
+import { Tv2SourceLayer } from '../value-objects/tv2-layers'
+import { PieceLifespan } from '../../../model/enums/piece-lifespan'
+import { TransitionType } from '../../../model/enums/transition-type'
+import { Tv2ActionContentType, Tv2RemoteAction } from '../value-objects/tv2-action'
+import {
+  Tv2AudioTimelineObjectFactory
+} from '../timeline-object-factories/interfaces/tv2-audio-timeline-object-factory'
+import { Tv2PieceType } from '../enums/tv2-piece-type'
+import {
+  Tv2VideoMixerTimelineObjectFactory
+} from '../timeline-object-factories/interfaces/tv2-video-mixer-timeline-object-factory'
+import { TimelineEnable } from '../../../model/entities/timeline-enable'
+
+export class Tv2RemoteActionFactory {
+
+  constructor(
+    private readonly videoMixerTimelineObjectFactory: Tv2VideoMixerTimelineObjectFactory,
+    private readonly audioTimelineObjectFactory: Tv2AudioTimelineObjectFactory
+  ) {}
+
+  public createRemoteActions(blueprintConfiguration: Tv2BlueprintConfiguration): Tv2RemoteAction[] {
+    return blueprintConfiguration.studio.SourcesRM
+      .map(source => this.createInsertRemoteAsNextAction(blueprintConfiguration, source))
+  }
+
+  private createInsertRemoteAsNextAction(configuration: Tv2BlueprintConfiguration, remoteSource: Tv2SourceMappingWithSound): Tv2RemoteAction {
+    const partId: string = `remoteInsertActionPart_${remoteSource._id}`
+    const remotePieceInterface: PieceInterface = this.createRemotePiece(configuration, remoteSource, partId)
+    const partInterface: PartInterface = this.createPartInterface(partId, remoteSource)
+    return {
+      id: `remoteAsNextAction_${remoteSource._id}`,
+      name: `LIVE ${remoteSource.SourceName}`,
+      description: `Insert LIVE ${remoteSource.SourceName} as next.`,
+      type: PartActionType.INSERT_PART_AS_NEXT,
+      data: {
+        partInterface: partInterface,
+        pieceInterfaces: [remotePieceInterface]
+      },
+      metadata: {
+        contentType: Tv2ActionContentType.REMOTE,
+        remoteNumber: remoteSource.SourceName,
+      },
+    }
+  }
+
+  private createRemotePiece(configuration: Tv2BlueprintConfiguration, source: Tv2SourceMappingWithSound, parentPartId: string): PieceInterface {
+    const remoteTimelineObjects: TimelineObject[] = this.createRemoteTimelineObjects(source)
+    const audioTimelineObjects: TimelineObject[] = this.audioTimelineObjectFactory.createTimelineObjectsForSource(configuration, source)
```

Comment:
It's a bit weird that you have a method called `createRemoteTimelineObjects()` but you have to create audioTimelineObjects right after calling it because `createRemoteTimelineObjects()` does not create all the timelineObjects for the Remote.

Either, you should move the creation of the audioTimelineObjects into `createRemoteTimelineObjects()` or you should rename `createRemoteTimelineObjects()` to `createVideoMixerTimelineObjects()` since that is what is does.

Reply:
I have opted for `createVideoMixerTimlineObjects`. It is also updated in the camera action factory.

Reply:
Nicely spotted for Cameras as well :)